arm/src/transaction.rs

```diff
@@ -100,6 +108,67 @@ impl Transaction {
     }
 }
 
+#[cfg(feature = "aggregation")]
+impl Transaction {
+    /// Aggregates all the transaction proofs with the default strategy.
+    pub fn aggregate(&mut self) -> Option<()> {
+        self.aggregate_with_strategy(AggregationStrategy::Batch)
+    }
+
+    /// Aggregates all the transaction proofs using the passed aggregation strategy.
+    /// If aggregation is successful, `self` contains an aggregation proof and its
+    /// compliance and logic proofs are set to `None`. Else proofs are untouched.
+    pub fn aggregate_with_strategy(&mut self, strategy: AggregationStrategy) -> Option<()> {
+        let agg_proof = match strategy {
+            AggregationStrategy::Sequential => {
+                SequentialAggregation::prove_transaction_aggregation(self)
+                    .map(AggregationProof::Sequential)
+            }
+            AggregationStrategy::Batch => {
+                BatchAggregation::prove_transaction_aggregation(self).map(AggregationProof::Batch)
+            }
+        };
+
+        if agg_proof.is_some() {
+            self.aggregation_proof = bincode::serialize(&agg_proof.unwrap()).ok();
+        }
+
+        if self.aggregation_proof.is_some() {
+            self.erase_base_proofs();
```

Comment:
I agree we should ideally remove original proofs after generating the aggregated proof. Let's discuss whether we need to keep them around, just in case PA requires them or if we’ll need them somewhere else. For now, I’ll just leave a note here.